tracing/tracing.go

```diff
@@ -6,51 +6,194 @@ package tracing
+		cfg.samplerRemoteURL = fmt.Sprintf("http://%s:%d/sampling", e, DefaultSamplingServerPort)
+	}
 
-	closer, err := cfg.InitGlobalTracer(serviceName, opts...)
+	cfg.customAttributes, err = parseAttributes(os.Getenv("JAEGER_TAGS"))
 	if err != nil {
-		return nil, errors.Wrap(err, "could not initialize jaeger tracer")
+		return nil, errors.Wrap(err, "could not parse JAEGER_TAGS")
 	}
-	return closer, nil
+	return &cfg, nil
 }
 
-// NewFromEnv is a convenience function to allow tracing configuration
-// via environment variables
-//
-// Tracing will be enabled if one (or more) of the following environment variables is used to configure trace reporting:
-// - JAEGER_AGENT_HOST
-// - JAEGER_SAMPLER_MANAGER_HOST_PORT
-func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, error) {
-	cfg, err := jaegercfg.FromEnv()
+func (cfg config) initJaegerTracerProvider(serviceName string) (*Closer, error) {
+	// Read environment variables to configure Jaeger
+	var ep jaegerotel.EndpointOption
+	// Create the jaeger exporter: address can be either agent address (host:port) or collector URL
+	if cfg.host != "" {
+		ep = jaegerotel.WithAgentEndpoint(jaegerotel.WithAgentHost(cfg.host), jaegerotel.WithAgentPort(strconv.Itoa(cfg.port)))
+	} else {
+		ep = jaegerotel.WithCollectorEndpoint(jaegerotel.WithEndpoint(cfg.collectorURL))
+	}
+	exp, err := jaegerotel.New(ep)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
+		return nil, err
 	}
 
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
-		return nil, ErrBlankTraceConfiguration
+	res, err := NewResource(serviceName, cfg.customAttributes)
+	if err != nil {
+		return nil, err
+	}
+
+	// Configure sampling strategy
+	sampler := tracesdk.AlwaysSample()
+	if cfg.samplerType == "const" || cfg.samplerType == "probabilistic" {
+		sampler = tracesdk.TraceIDRatioBased(cfg.samplerParam)
+	} else if cfg.samplerType == "remote" {
+		sampler = jaegerremote.New(serviceName, jaegerremote.WithSamplingServerURL(cfg.samplerRemoteURL),
+			jaegerremote.WithInitialSampler(tracesdk.TraceIDRatioBased(cfg.samplerParam)))
+	} else if cfg.samplerType != "" {
+		return nil, errors.Errorf("unknown sampler type %q", cfg.samplerType)
 	}
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithBatcher(exp),
+		tracesdk.WithResource(res),
+		tracesdk.WithSampler(sampler),
+	)
+
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator([]propagation.TextMapPropagator{
+		// w3c Propagator is the default propagator for opentelemetry
+		propagation.TraceContext{}, propagation.Baggage{},
+		// jaeger Propagator is for opentracing backwards compatibility
+		jaegerpropagator.Jaeger{},
+	}...))
+
+	// OpenTracing <=> OpenTelemetry bridge
+	// The tracer name is empty so that the bridge uses the default tracer.
+	otTracer, _ := bridge.NewTracerPair(tp.Tracer(""))
```

Comment:
Does it *have* to be empty? If possible at all, I'd use this repo's name in the tracer.

Reply:
@gouthamve Could I set Tracer name to dskit, and do tp.GetTracer("") to have the correct tracer cfg with otel sdk?